### PR TITLE
TLS SubjectAltName sticky buffer/v6

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/5234

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1743

Previous PR: https://github.com/OISF/suricata/pull/10750

Changes since v5:
- comments addressed
- doc updates
- logging added
- schema updated

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW2_autofp_suri_time.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_autofp_stats_chk | | |
| .uptime | 138 | 147 | 106.52% | 

Pipeline 19920

File: etc/schema.json [hunk 1]
```diff
@@ -3733,7 +3735,8 @@
                     }
                 },
                 "memcap_pressure": {
-                    "description": "Percentage of memcaps used by flow, stream, stream-reassembly and app-layer-http",
+                    "description":
+                            "Percentage of memcaps used by flow, stream, stream-reassembly and app-layer-http",
```
Comment (jufajardini) [inline etc/schema.json:3739]: nit: clang formatting changes unrelated to schema additions?

File: etc/schema.json [hunk 2]
```diff
@@ -5484,6 +5490,12 @@
                         "issuerdn": {
                             "type": "string"
                         },
+                        "subjectaltname": {
+                            "type": "array",
+                            "items": {
+                                "type": "string"
```
Comment (jufajardini) [inline etc/schema.json:5496]: nit: wouldn't it be a good time to add a description for the field, already?

File: src/detect-tls-subjectaltname.c [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+} PrefilterMpmTlsSubjectAltName;
+
+/**
+ * \brief Registration function for keyword: tls.subjectaltname
+ */
+void DetectTlsSubjectAltNameRegister(void)
+{
+    sigmatch_table[DETECT_AL_TLS_SUBJECTALTNAME].name = "tls.subjectaltname";
+    sigmatch_table[DETECT_AL_TLS_SUBJECTALTNAME].desc =
+            "sticky buffer to match the TLS Subject Alternative Name buffer";
```
Comment (jufajardini) [inline src/detect-tls-subjectaltname.c:76]: Wondering: would it be crazy or too difficult to maybe have an automated way of picking this info to help populate the json schema, for keywords?

**Review by jufajardini — COMMENTED** [review]
Doc changes look good to me, comments left are nit that shouldn't block merge, and one is just a possibly crazy idea :P

File: etc/schema.json (hunk 1 above)
Comment (inashivb) [reply etc/schema.json:3739]: indeed. have done this so many times separately in the recent PRs 😩 got lazy. Thanks, will fix

File: etc/schema.json (hunk 2 above)
Comment (inashivb) [reply etc/schema.json:5496]: absolutely. Thank you!

File: src/detect-tls-subjectaltname.c (hunk 3 above)
Comment (inashivb) [reply src/detect-tls-subjectaltname.c:76]: We'll have to at least write one more script to direct the output from `list-keywords` output.
Or, there could be a logging fn in registration to take care of all logging related work. Not sure about this one if it's feasible.
But, nice thought. :) Out of scope of this PR though. But, maybe good to pitch to others and take their opinion.

**Pull request closed without merging**